### PR TITLE
JSTOR service tidy up

### PR DESCRIPTION
**Pull request by jon-betts**
For:

 * https://github.com/hypothesis/via/issues/734

This is a bit of ground clearing before working on the JSTOR pass through work.

### CODE REVIEW

File: tests/unit/services.py [hunk 1]
```diff
@@ -50,4 +50,7 @@ def pdf_url_builder_service(mock_service):
 
 @pytest.fixture
 def jstor_api(mock_service):
-    return mock_service(JSTORAPI)
+    jstor_api = mock_service(JSTORAPI)
+
+    jstor_api.is_jstor_url.return_value = False
```
Comment (jon-betts) [inline tests/unit/services.py:55]: We use this to check if things are JSTOR URLs indirectly in a lot of tests. By default mocks evaluate to true, and therefore _everything_ becomes a JSTOR URL which is likely not the intention of most of the tests.

File: tests/unit/via/services/pdf_url_test.py [hunk 2]
```diff
@@ -91,32 +91,25 @@ def test_google_file_url(
             ),
         ),
     )
-    def test_onedrive_url(
-        self, svc, secure_link_service, google_drive_api, url, endpoint_url
-    ):
-        google_drive_api.parse_file_url.return_value = None
```
Comment (jon-betts) [inline tests/unit/via/services/pdf_url_test.py:97]: Moved this repeated setup to the end

File: tests/unit/via/services/jstor_test.py [hunk 3]
```diff
@@ -8,30 +8,50 @@
...
-        assert isinstance(factory(sentinel.context, pyramid_request), JSTORAPI)
+    def test_it(self, pyramid_request, JSTORAPI, http_service):
+        pyramid_request.registry.settings["jstor_pdf_url"] = sentinel.jstor_pdf_url
+
+        svc = factory(sentinel.context, pyramid_request)
+
+        JSTORAPI.assert_called_once_with(
+            http_service=http_service, api_url=sentinel.jstor_pdf_url
+        )
+        assert svc == JSTORAPI.return_value
```
Comment (jon-betts) [inline tests/unit/via/services/jstor_test.py:53]: This didn't really assert anything about how the object was created before

File: via/services/jstor.py [hunk 4]
```diff
@@ -3,23 +3,39 @@
 
 
 class JSTORAPI:
+    """An interface for dealing with JSTOR documents."""
+
     DEFAULT_DOI_PREFIX = "10.2307"
-    """Main DOI prefix. Used for real DOIs and non registered pseudo-DOIs with the same format"""
+    """Used when no DOI prefix can be found."""
```
Comment (jon-betts) [inline via/services/jstor.py:9]: I don't understand what this was trying to say before, so I may have phrased this wrong.

File: via/services/jstor.py [hunk 5]
```diff
@@ -3,23 +3,39 @@
...
+        return url.startswith("jstor://")
+
+    def stream_pdf(self, url, jstor_ip):
+        """Get a stream for the given JSTOR url.
+
+        :param url: The URL to stream
+        :param jstor_ip: The IP we use to authenticate ourselves with
+        """
 
-    def jstor_pdf_stream(self, jstor_url: str, jstor_ip: str):
```
Comment (jon-betts) [inline via/services/jstor.py:17]: Turned this noun name into more of a verb

File: via/services/jstor.py (hunk 4 above)
Comment (marcospri) [reply via/services/jstor.py:9]: Yeah this is clearer. As far as I remember DOI without a prefix was not a real DOI but something else. That distinction is completely irrelevant.

**Review by marcospri — APPROVED** [review]

**Pull request merged**